### DISCUSSION TITLE
Speed/ Frequency of Reading

### DISCUSSION THREAD

**Discussion by VirtualAI** [jomjol/AI-on-the-edge-device#938] [General]
Hello,

how ist it possible to absolute speed up the time between frames?
At the moment, i detected some paramters to gain speed, but how to come do the absolute max?

I found image quality (What is the minimum needed image quality?)
Fixed Exposure = true
Search Field (what is  the smallest working setting?)
AlignmentAlgo =  Fast
WaitBeforeTakingPicture = 0 (The digits are illuminated by them self)

Is there anything more to speed up?

Best regards,
Michael

**Comment by jomjol** [comment]
You can use a dedicated tflite, which minimum parameters (must be trained personally).
Search field depends on your mechanical stability. If you can ensure a mechanical alignment down to 1 pixel, you can make it very small.

But if you are searching for a very fast system, this is not the right one, as it is limited by the esp32 calculation power.

**Reply by VirtualAI** [reply]
Thanks for the fast reply. I will test out minimum value regarding to mechanical stability.
If got a further problem. My digits are illuminated, i have tried different alignements settings, but still no good result. (See image attached)
![image](https://user-images.githubusercontent.com/13152991/186667937-a0ac6097-c146-4c16-a48f-c113b595872a.png)

Any ideas/hints?

Thank you very much

**Reply by jomjol** [reply]
First of all they don't look sharp and secondly they are most probably not part of the current training set. 

Regarding speed I have a commercial version, that can go down to faster than 2s on an ESP32. But it needs some dedicated settings.
